python/ray/_private/gc_collect_manager.py

```diff
@@ -0,0 +1,64 @@
+import gc
+import logging
+import threading
+import time
+from typing import Callable, Optional
+
+logger = logging.getLogger(__name__)
+
+
+class PythonGCThread(threading.Thread):
+    """A background thread that triggers Python garbage collection.
+    This thread waits for GC events from CoreWorker and triggers `gc.collect()` when
+    requested, ensuring that collections are spaced out by at least
+    `min_interval` seconds."""
+
+    def __init__(
+        self, min_interval: int = 5, gc_collect_func: Optional[Callable] = None
```

Comment:
```suggestion
        self, *, min_interval_s: int = 5, gc_collect_func: Optional[Callable] = None
```
nits:

- force args to be passed as kwargs
- use unit in the arg name for clarity